lib/ansible/modules/windows/win_initialize_disk.ps1

```diff
@@ -0,0 +1,151 @@
+#!powershell
+
+# Copyright: (c) 2019, Brant Evans <bevans@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -OSVersion 6.2
+
+Set-StrictMode -Version 2
+
+$spec = @{
+    options = @{
+        disk_number = @{ type = "int" }
+        uniqueid = @{ type = "str" }
+        path = @{ type = "str" }
+        style = @{ type = "str"; choices = "gpt", "mbr"; default = "gpt" }
+        online = @{ type = "bool"; default = $true }
+        force = @{ type = "bool"; default = $false }
+    }
+    mutually_exclusive = @(
+        ,@('disk_number', 'uniqueid', 'path')
+    )
+    required_one_of = @(
+        ,@('disk_number', 'uniqueid', 'path')
+    )
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$disk_number = $module.Params.disk_number
+$uniqueid = $module.Params.uniqueid
+$path = $module.Params.path
+$partition_style = $module.Params.style
+$bring_online = $module.Params.online
+$force_init = $module.Params.force
+
+function Get-AnsibleDisk {
```

Comment:
I think you should allow for this function to get and return lists, i.e if i specify one disk by ID and another with a number - I want to Init both eventually

Reply:
Allowing for lists would make the behavior of this module different from how the `win_format` and `win_partition` modules work (as well as the Unix equivalent modules such as `filesystem`).

The loop structure can be used on the task to provide a list with multiple disks and identifications. For example:

```yaml
  vars:
    - disks:
        - diskNumber: 1
          style: gpt
        - uniqueId: "60022480622E439002680C84CD374969"
        - path: '\\?\scsi#disk&ven_msft&prod_virtual_disk#2&1f4adffe&0&000001#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}'
          style: mbr

  tasks:
    - name: Initialize Disks
      win_initialize_disk:
        disk_number: '{{ item.diskNumber | default(omit) }}'
        uniqueid: '{{ item.uniqueId | default(omit) }}'
        path: '{{ item.path | default(omit) }}'
        style: '{{ item.style | default(omit) }}'
      loop: '{{ disks }}'

```

This also has the advantage that the other parameters can be specified differently for each disk.

Maybe this is a good candidate as an example in the documentation?